stdlib/regexp/regexp.go

```diff
@@ -0,0 +1,42 @@
+package regexp
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+var SpecialFns map[string]values.Function
+
+func init() {
+
+	SpecialFns = map[string]values.Function{
+		"compileRegexp": values.NewFunction(
+			"compileRegexp",
+			semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+				Parameters: map[string]semantic.PolyType{"v": semantic.String},
+				Required:   semantic.LabelSet{"v"},
+				Return:     semantic.Regexp,
+			}),
+			func(args values.Object) (values.Value, error) {
+				v, ok := args.Get("v")
+				if !ok {
+					return nil, errors.New("missing argument v")
+				}
+
+				if v.Type().Nature() == semantic.String {
+					re, ok := regexp.Compile(v.Str())
+					return values.NewRegexp(re), ok
```

Comment:
for errors, we usually use a variable `err`  unless it's already been used.   `ok` is used as a name when the function returns a boolean check instead of an error type.  

more important:   check the error before you call `values.NewRegexp`  otherwise, you may get bad things happening if `re` is null.